app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -290,6 +302,94 @@ private void selectPointer() {
+                    .setTitle(R.string.edit_message)
+                    .setView(dialogView)
+                    .setPositiveButton(android.R.string.ok, null)
+                    .setNegativeButton(android.R.string.cancel, null)
+                    .create();
+            dialog.getWindow().setSoftInputMode(
+                    WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
+            dialog.show();
+
+            final EditText question = (EditText) dialogView.findViewById(R.id.quiz_question);
+            question.setText(message.getContent().trim());
+
+            //Move cursor to end of text
+            question.setSelection(question.getText().length());
+
+            //OK button listener
+            dialog.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
+                //Show edit option only on if current user send it.
+                @Override
+                public void onClick(View view) {
+                    dialog.cancel();
+                    InputMethodManager imm = (InputMethodManager) getActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
+                    imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
+                    //Start a progress dialog indicating editing message
+                    final ProgressDialog progress = new ProgressDialog(getActivity());
+                    progress.setCancelable(false);
+                    progress.setMessage(ZulipApp.get().getString(R.string.editing_message));
+                    progress.show();
+                    final String editedMessageContent = question.getText().toString();
+
+                    AsyncEdit sender = new AsyncEdit(((ZulipActivity) getActivity()), message, editedMessageContent);
+                    sender.setCallback(new ZulipAsyncPushTask.AsyncTaskCompleteListener() {
+                        public void onTaskComplete(String result, JSONObject jsonObject) {
+                            message.setContent(editedMessageContent);
+                            message.setFormattedContent(editedMessageContent);
+                            adapter.notifyItemChanged(position);
+                            progress.dismiss();
+                            Toast.makeText(getActivity(), R.string.message_edited, Toast.LENGTH_SHORT).show();
+                        }
+
+                        public void onTaskFailure(String result) {
+                            progress.dismiss();
+                            showEditFailureToast(result);
+                        }
+                    });
+                    sender.execute();
+                }
+            });
+        }
+    }
+
+    /**
+     * Shows a error {@link Toast} corresponding to server result
+     *
+     * @param result Server edit failure response as string
+     */
+    private void showEditFailureToast(String result) {
+        try {
+            JSONObject jsonObject = new JSONObject(result);
```

Comment:
@kunall17 Thanks for reviewing. Actually we don't need this method anymore because `Time limit error` from server is already being handled now.